mobile/src/main/java/org/openhab/habdroid/ui/ItemPickerAdapter.java

```diff
@@ -0,0 +1,174 @@
+        }
+        notifyDataSetChanged();
+    }
+
+    private class ItemNameComparator implements Comparator<Item> {
+        public int compare(Item left, Item right) {
+            return left.name().compareToIgnoreCase(right.name());
+        }
+    }
+
+    public void clear() {
+        mItems.clear();
+        notifyDataSetChanged();
+    }
+
+    public int findPositionForName(String name) {
+        for (int i = 0; i < mItems.size(); i++) {
+            if (TextUtils.equals(mItems.get(i).name(), name)) {
+                return i;
+            }
+        }
+        return -1;
+    }
+
+    public void highlightItem(int position) {
+        mHighlightedPosition = position;
+        notifyItemChanged(position);
+    }
+
+    @NonNull
+    @Override
+    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        return new ItemViewHolder(mInflater, parent);
+    }
+
+    @Override
+    public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
+        ItemViewHolder itemViewHolder = (ItemViewHolder) holder;
+        itemViewHolder.bind(mItems.get(position));
+        holder.itemView.setOnClickListener(mItemClickListener != null ? this : null);
+
+        if (position == mHighlightedPosition) {
+            final View v = holder.itemView;
+            v.post(() -> {
+                if (v.getBackground() != null) {
+                    final int centerX = v.getWidth() / 2;
+                    final int centerY = v.getHeight() / 2;
+                    DrawableCompat.setHotspot(v.getBackground(), centerX, centerY);
+                }
+                v.setPressed(true);
+                v.setPressed(false);
+                mHighlightedPosition = -1;
+            });
+        }
+    }
+
+    @Override
+    public int getItemCount() {
+        return mItems.size();
+    }
+
+    public static class ItemViewHolder extends RecyclerView.ViewHolder {
+        final TextView mItemNameView;
+        final TextView mItemLabelView;
+        final WidgetImageView mIconView;
+        final TextView mItemTypeView;
+
+        public ItemViewHolder(LayoutInflater inflater, ViewGroup parent) {
+            super(inflater.inflate(R.layout.itempickerlist_item, parent, false));
+            mItemNameView = itemView.findViewById(R.id.itemName);
+            mItemLabelView = itemView.findViewById(R.id.itemLabel);
+            mItemTypeView = itemView.findViewById(R.id.itemType);
+            mIconView = itemView.findViewById(R.id.itemIcon);
+            itemView.setTag(this);
+        }
+
+        public void bind(Item item) {
+            mItemNameView.setText(item.name());
+            mItemLabelView.setText(item.label());
+            mItemTypeView.setText(item.type().toString());
+
+            Connection connection;
+            try {
+                connection = ConnectionFactory.getUsableConnection();
+            } catch (ConnectionException e) {
+                connection = null;
+            }
```

Comment:
We should probably introduce a `getUsableConnectionOrNull` method to `ConnectionFactory` to avoid the throwing overhead when we're not interested in the actual exception. I'll do this in the Kotlin conversion.